cypress/specs/acceptance/pages/view.spec.ts

```diff
@@ -0,0 +1,96 @@
+import Chance from 'chance';
+import { getTranslation, randomLocale } from '@/support';
+
+describe('View Secret page', () => {
+  const chance = new Chance();
+  let locale: string;
+  const id = chance.guid();
+
+  const visit = (): void => {
+    cy.visit(`/${locale}/${id}`);
+    cy.injectAxe();
+  };
+
+  beforeEach(() => {
+    locale = randomLocale(false);
+    cy.viewport(chance.pickone(['iphone-x', 'macbook-13', 'samsung-s10']));
+  });
+
+  it('should render important callouts and labels', () => {
+    // set intercept before the page visit
+    cy.intercept('GET', `/api/secrets/${id}/details`, {
+      statusCode: 200,
+    });
+    visit();
+    cy.contains('h2', getTranslation(locale, 'secrets.view.title'));
+    cy.contains('button', getTranslation(locale, 'secrets.view.button'));
+    cy.assertSecurityInfo(locale);
+  });
+
+  it('should handle 404 status code response from the API', () => {
+    // set intercept before the page visit
+    cy.intercept('GET', `/api/secrets/${id}/details`, {
+      statusCode: 404,
+    });
+    visit();
+    // temporarily disable halting on uncaught client exceptions (used to redirect)
+    cy.on('uncaught:exception', () => false);
+    cy.assert404(locale);
+  });
+
+  it('should handle 500 status code response from the API', () => {
+    // set intercept before the page visit
+    cy.intercept('GET', `/api/secrets/${id}/details`, {
+      statusCode: 500,
+    });
+    visit();
+    // temporarily disable halting on uncaught client exceptions (used to redirect)
+    cy.on('uncaught:exception', () => false);
```

Comment:
Does this not need to be before the call to `visit()` which generates the uncaught client exception?

Reply:
It is safer to move before the `visit` statement. The exception is thrown by the `useApi` hook inside the page, that only runs after the page is completely loaded.